### PR TITLE
Made Random Number Generator Guesser Game in Rust

### PR DESCRIPTION
**Pull request by lucasod3v**
I am a beginner so the code might be bad...

### CODE REVIEW

**Review by StoneMcYT — COMMENTED** [review]
LGTM also can you add into the readme.md what is rng_guesser for and who made it that kind of thing?

**Comment by ghost** [comment]
Read rust code guidelines please.

**Review by ghost — APPROVED** [review]
Good to go(tip: next time read code guidelines)

**Review by MrSerge01 — APPROVED** [review]
nice
Screenshot_1048
anwyays, idk anything about rust but if Akane told LGTM, then

LGTM,
-The Goos.

**Comment by lucasod3v** [comment]
> Good to go(tip: next time read code guidelines)

Thx for the fix, have to fix my formatter, it doesn't like it in {}.

File: rng_guesser/number_guessing_game/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    println!("{start_txt}");
+    println!("Made by lucasodev - UNLICENSE");
+    println!(" ");
+    println!(" ");
+    
+    guessing();
+}
+
+fn guessing() {
+    let random_number = get_random_number();
```
Comment (Sammy99jsp) [inline rng_guesser/number_guessing_game/src/main.rs]: Possible variable naming, would something like `target` or `target_number` be clearer ?

Comment (lucasod3v) [reply rng_guesser/number_guessing_game/src/main.rs]: Yes, I can fix it when I come back to my pc.

**Comment by lucasod3v** [comment]
> Good to go(tip: next time read code guidelines)

image

This is the Rust code guidelines. There are no guidelines...lol.

File: rng_guesser/number_guessing_game/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    
+    let mut guess_right: bool = false;
```
Comment (Sammy99jsp) [inline rng_guesser/number_guessing_game/src/main.rs]: Consider the fact that `guess_right` is actually just the result of the comparison `guess == random_number`.

So, if the code was re-written to something like this:
```rust
loop { // loop {} in rust is just a shorthand for `while true {....}`
    let guess = get_user_input();
    if guess == target {
        println!("Yeah, it was right, the number was {target}!");
        return; // By returning here, we immediately get out of the loop and `guessing()`, so no need for an `else` block.
    } 
    println!("Sorry, Your answer was wrong!");
}
```

there isn't a need for the `guess_right` variable.

Comment (lucasod3v) [reply rng_guesser/number_guessing_game/src/main.rs]: Ohh, yeah, I didn't know that. I have just programmed in rust for two days... will fix it later.

**Comment by ghost** [comment]
> > Good to go(tip: next time read code guidelines)
> 
> image
> 
> This is the Rust code guidelines. There are no guidelines...lol.

https://github.com/Avdan-OS/docsite/blob/dev/docs/avdanos/contributing/development/code_guidelines/rust_guidelines.mdx

File: rng_guesser/number_guessing_game/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            guess_right = true;
+        } else {
+            println!("Sorry, Your answer was wrong!");
+            continue;
+        }
+    }
+}
+
+fn get_random_number() -> i64 {
+    let _random_number: i64 = thread_rng().gen_range(0, 11);
```
Comment (Sammy99jsp) [inline rng_guesser/number_guessing_game/src/main.rs]: Unnecessary variable `_random_number` :

You could re-write this into:
```rust
fn get_random_number() -> i64 {
    thread_rng().gen_range(0, 11)
}
```

Comment (lucasod3v) [reply rng_guesser/number_guessing_game/src/main.rs]: Yeah

File: rng_guesser/number_guessing_game/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+fn get_user_input() -> i64 {
+    println!("Please input a guessing number under (Must be between 0 - 10):");
+    let mut guess = String::new();
+    io::stdin()
+        .read_line(&mut guess)
+        .expect("Error while reading line!");
+
+    let _int_guess: i64 = guess.trim().parse().unwrap();
```
Comment (Sammy99jsp) [inline rng_guesser/number_guessing_game/src/main.rs]: Same thing as above here:
```rust
fn get_user_input() -> i64 {
    // ...
    guess.trim().parse().unwrap()
}
```

**Review by Golem642 — COMMENTED** [review]
I mean i don't know rust but from what I can understand this seems good

**Pull request merged**